README.md

```diff
@@ -1,22 +1,59 @@
 # Sparse Vector (SparseVec)
 
-A SparseVec compresses a matrix (passed as a one dimensional vector) using row
-displacement as described in [1]. It then uses a PackedVec to further
-reduce the size of the result.
+A SparseVec efficiently encodes a two-dimensional matrix of integers. The input
+matrix must be encoded as a one-dimensional vector of integers with a
+row-length. Given an empty value, the SparseVec uses row displacement as
+described in [1] for the compression and encodes the result further using a
+PackedVec.
 
-## Usage
+[1] Tarjan, Robert Endre, and Andrew Chi-Chih Yao. "Storing a sparse table."
+Communications of the ACM 22.11 (1979): 606-611.
 
-```
+# Usage
+
+```rust
 extern crate sparse_vec;
 use sparse_vec::SparseVec;
 
 fn main() {
-    let v:Vec<usize> = vec![1,2,3,4,5,6,7,8];
+    use sparsevec::SparseVec;
+    let v:Vec<usize> = vec![1,0,0,0,
+                            0,0,7,8,
+                            9,0,0,3];
     let sv = SparseVec::from(&v, 0, 4);
+    assert_eq!(sv.get(0,0).unwrap(), 1);
     assert_eq!(sv.get(1,2).unwrap(), 7);
+    assert_eq!(sv.get(2,3).unwrap(), 3);
 }
-
 ```
 
-[1] Tarjan, Robert Endre, and Andrew Chi-Chih Yao. "Storing a sparse table."
-Communications of the ACM 22.11 (1979): 606-611.
+# How it works
+
+Let's take as an example the two-dimensional vector
+```
+1 0 0
+2 0 0
+3 0 0
+0 0 4
+```
+represented as a one dimensional vector `v = [1,0,0,2,0,0,3,0,0,0,0,4]` with row-length 3.
+Storing this vector in memory is wasteful as the majority of its elements is 0. We can compress
+this vector using row displacement, which merges all rows into a vector such that no two
```

Comment:
Can we say something like "such that non-zero entries are never mapped to the same position"? ["No two non-zero" is a bit hard to parse at first.]